### PR TITLE
Implementation of sendTempPacket and sendVoltagePacket

### PR DESCRIPTION
**Pull request by Rolandw31**

### CODE REVIEW

File: Core/Src/cells.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    static uint32_t idTracker = HVC_VCU_CELL_VOLTAGES_START;
+    if(idTracker == HVC_VCU_CELL_VOLTAGES_START + 280) {
+        idTracker = HVC_VCU_CELL_VOLTAGES_START;
+    }
+
+    // convert voltageData from float in mV to uint8_t that uses 2 bytes per voltage
+    static uint8_t voltages[8];
+    static int offset = (int)(idTracker - HVC_VCU_CELL_VOLTAGES_START) / 2;
+
+    for(int i = offset ; i < (offset + 4) ; i++) {
```
Comment (matt-mekha) [inline Core/Src/cells.cpp:77]: change this to `i=0; i<4`

Comment (matt-mekha) [inline Core/Src/cells.cpp:77]: change this to `i=0; i<4`

File: Core/Src/cells.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        auto value = (uint16_t) (voltageData[i] / 0.001f);
+        auto byte1 = static_cast<uint8_t>((value & 0xFF00) >> 8);
+        auto byte2 = static_cast<uint8_t>(value & 0x00FF);
+
+        voltages[i] = byte1;
```
Comment (matt-mekha) [inline Core/Src/cells.cpp:82]: `i*2`

File: Core/Src/cells.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    if(idTracker == HVC_VCU_CELL_VOLTAGES_START + 280) {
+        idTracker = HVC_VCU_CELL_VOLTAGES_START;
+    }
+
+    // convert voltageData from float in mV to uint8_t that uses 2 bytes per voltage
+    static uint8_t voltages[8];
+    static int offset = (int)(idTracker - HVC_VCU_CELL_VOLTAGES_START) / 2;
+
+    for(int i = offset ; i < (offset + 4) ; i++) {
+        auto value = (uint16_t) (voltageData[i] / 0.001f);
```
Comment (matt-mekha) [inline Core/Src/cells.cpp:78]: incorporate the offset here instead of in the for loop

File: Core/Src/contactors.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#include "../Inc/contactors.h"
+#include "main.h"
+
+void setPrechargeContactor(bool on) {
+    HAL_GPIO_WritePin(Precharge_Enable_GPIO_Port, Precharge_Enable_Pin, (GPIO_PinState) on);
+}
+
+void setDriveContactor(bool on) {
+    HAL_GPIO_WritePin(Close_HV__Signal_GPIO_Port, Close_HV__Signal_Pin, (GPIO_PinState) on);
+    HAL_GPIO_WritePin(Close_HV__SignalC5_GPIO_Port, Close_HV__SignalC5_Pin, (GPIO_PinState) on);
```
Comment (matt-mekha) [inline Core/Src/contactors.cpp:14]: why are there two pins? what's the difference?

File: Core/Src/imd.cpp [hunk 5]
```diff
@@ -0,0 +1,15 @@
+//
+// Created by rolandwang on 11/12/2023.
+//
+
+#include "../Inc/imd.h"
+#include "main.h"
+
+bool isImdOk() {
+    if(HAL_GPIO_ReadPin(IMD_Data_GPIO_Port, IMD_Data_Pin)) {
```
Comment (matt-mekha) [inline Core/Src/imd.cpp:9]: make one-liner with `HAL_GPIO_ReadPin(...) == GPIO_PIN_SET`

**Review by matt-mekha — CHANGES_REQUESTED** [review]
Overall looks good, some minor things to change

File: Core/Src/cells.cpp (hunk 1 above)
Comment (Rolandw31) [reply Core/Src/cells.cpp:77]: changes made

File: Core/Src/cells.cpp (hunk 2 above)
Comment (Rolandw31) [reply Core/Src/cells.cpp:82]: changes made

File: Core/Src/contactors.cpp (hunk 4 above)
Comment (Rolandw31) [reply Core/Src/contactors.cpp:14]: There is a positive and negative contactor that we are controlling, the port/pin names need some renaming I think.

File: Core/Src/cells.cpp (hunk 3 above)
Comment (Rolandw31) [reply Core/Src/cells.cpp:78]: changes made

**Review by matt-mekha — APPROVED** [review]

**Pull request merged**